lib/games_manager.rb

```diff
@@ -171,65 +171,43 @@ def get_season_games(season)
       game.game_id
     end
   end
-
+  
   def total_home_goals
-    home_team_id_goals = {}
-    games.each do |game|
-      if home_team_id_goals[game.home_team_id].nil?
-        home_team_id_goals[game.home_team_id] = game.home_goals
-      else
-        home_team_id_goals[game.home_team_id] += game.home_goals
-      end
+    team_id_by_home_goals = games.map do |game|
+      [game.home_team_id, game.home_goals]
     end
-    home_team_id_goals
+    sum_values(team_id_by_home_goals)
   end
-
-  def total_home_games
-    home_team_id_games = {}
-    games.each do |game|
-      if home_team_id_games[game.home_team_id].nil?
-        home_team_id_games[game.home_team_id] = 1
-      else
-        home_team_id_games[game.home_team_id] += 1
-      end
+  
+  def total_away_goals
+    team_id_by_away_goals = games.map do |game|
+      [game.away_team_id, game.away_goals]
     end
-    home_team_id_games
+    sum_values(team_id_by_away_goals)
   end
-
-  def highest_scoring_home
-    averages = total_home_goals.merge(total_home_games) do |team_id, goals, games|
-      (goals/games.to_f).round(2)
-    end
-    average_max = averages.max_by do |team_id, average|
-      average
+  
+  def total_home_games
+    team_id_by_home_games = games.map do |game|
+      [game.home_team_id, 1]
     end
-     average_max[0]
+    sum_values(team_id_by_home_games)
   end
-
-  def total_away_goals
-    visitor_team_id_goals = {}
-    games.each do |game|
-      if visitor_team_id_goals[game.away_team_id].nil?
-        visitor_team_id_goals[game.away_team_id] = game.away_goals
-      else
-        visitor_team_id_goals[game.away_team_id] += game.away_goals
-      end
+  
+  def total_away_games
+    team_id_by_away_games = games.map do |game|
+      [game.away_team_id, 1]
     end
-    visitor_team_id_goals
+    sum_values(team_id_by_away_games)
   end
-
-  def total_away_games
-    visitor_team_id_games = {}
-    games.each do |game|
-      if visitor_team_id_games[game.away_team_id].nil?
-        visitor_team_id_games[game.away_team_id] = 1
-      else
-        visitor_team_id_games[game.away_team_id] += 1
-      end
+  
+  def sum_values(key_value_arr)
+    sum = Hash.new(0)
+    key_value_arr.each do |key, value|
```

Comment:
I am slightly confused about the argument key_value_arr. 
are passing in an array with two values eg. [index0, index1] where array[0] becomes a key, and array[1] becomes the value
or are you passing in a hash and then iterating through that hash, using the each method?